source/extensions/quic_listeners/quiche/envoy_quic_server_session.cc

```diff
@@ -0,0 +1,179 @@
+                                          stream_helper());
+}
+
+quic::QuicSpdyStream* EnvoyQuicServerSession::CreateIncomingStream(quic::QuicStreamId id) {
+  if (!ShouldCreateIncomingStream(id)) {
+    return nullptr;
+  }
+  auto stream = new EnvoyQuicServerStream(id, this, quic::BIDIRECTIONAL);
+  ActivateStream(absl::WrapUnique(stream));
+  setUpRequestDecoder(*stream);
+  return stream;
+}
+
+quic::QuicSpdyStream* EnvoyQuicServerSession::CreateIncomingStream(quic::PendingStream* pending) {
+  auto stream = new EnvoyQuicServerStream(pending, this, quic::BIDIRECTIONAL);
+  ActivateStream(absl::WrapUnique(stream));
+  setUpRequestDecoder(*stream);
+  return stream;
+}
+
+quic::QuicSpdyStream* EnvoyQuicServerSession::CreateOutgoingBidirectionalStream() {
+  // Not allow server initiated stream.
+  NOT_REACHED_GCOVR_EXCL_LINE;
+}
+
+quic::QuicSpdyStream* EnvoyQuicServerSession::CreateOutgoingUnidirectionalStream() {
+  NOT_REACHED_GCOVR_EXCL_LINE;
+}
+
+void EnvoyQuicServerSession::setUpRequestDecoder(EnvoyQuicStream& stream) {
+  ASSERT(http_connection_callbacks_ != nullptr);
+  Http::StreamDecoder& decoder = http_connection_callbacks_->newStream(stream);
+  stream.setDecoder(decoder);
+}
+
+void EnvoyQuicServerSession::OnConnectionClosed(const quic::QuicConnectionCloseFrame& frame,
+                                                quic::ConnectionCloseSource source) {
+  quic::QuicServerSessionBase::OnConnectionClosed(frame, source);
+  for (auto callback : network_connection_callbacks_) {
+    // Tell filters about connection close.
+    callback->onEvent(source == quic::ConnectionCloseSource::FROM_PEER
+                          ? Network::ConnectionEvent::RemoteClose
+                          : Network::ConnectionEvent::LocalClose);
+  }
+  transport_failure_reason_ = absl::StrCat(quic::QuicErrorCodeToString(frame.quic_error_code),
+                                           " with details: ", frame.error_details);
+}
+
+void EnvoyQuicServerSession::Initialize() {
+  quic::QuicServerSessionBase::Initialize();
+  reinterpret_cast<EnvoyQuicConnection*>(connection())->setEnvoyConnection(*this);
+}
+
+void EnvoyQuicServerSession::addWriteFilter(Network::WriteFilterSharedPtr filter) {
+  filter_manager_.addWriteFilter(filter);
+}
+
+void EnvoyQuicServerSession::addFilter(Network::FilterSharedPtr filter) {
+  filter_manager_.addFilter(filter);
+}
+
+void EnvoyQuicServerSession::addReadFilter(Network::ReadFilterSharedPtr filter) {
+  filter_manager_.addReadFilter(filter);
+}
+
+bool EnvoyQuicServerSession::initializeReadFilters() {
+  return filter_manager_.initializeReadFilters();
+}
+
+void EnvoyQuicServerSession::addConnectionCallbacks(Network::ConnectionCallbacks& cb) {
+  network_connection_callbacks_.push_back(&cb);
+}
+
+void EnvoyQuicServerSession::addBytesSentCallback(Network::Connection::BytesSentCb /*cb*/) {
+  // Proxy is not supported.
+  NOT_IMPLEMENTED_GCOVR_EXCL_LINE;
```

Comment:
I'm a bit twitchy about runtime crash because "we do not implement this part of the interface"
Maybe just a TODO to audit these out before unhiding the config?